lib/Server.js

```diff
@@ -3254,9 +3254,36 @@ class Server {
    * @param {string | string[]} watchPath
    * @param {WatchOptions} [watchOptions]
    */
-  watchFiles(watchPath, watchOptions) {
+  watchFiles(watchPath, watchOptions = {}) {
     const chokidar = require("chokidar");
-    const watcher = chokidar.watch(watchPath, watchOptions);
+    const isGlob = require("is-glob");
+
+    const watchPathArr = Array.isArray(watchPath) ? watchPath : [watchPath];
+    const watchPathGlobs = watchPathArr.filter((p) => isGlob(p));
+
+    // No need to do all this work when no globs are used
+    if (watchPathGlobs.length > 0) {
+      const globParent = require("glob-parent");
+      const picomatch = require("picomatch");
+
+      watchPathGlobs.forEach((p) => {
+        watchPathArr[watchPathArr.indexOf(p)] = globParent(p);
+      });
+
+      const matcher = picomatch(watchPathGlobs);
```

Comment:
What do you think about passing parameters here? Before we have them in watchOptions for globs

Also why do not use `tinyglobby`?

Reply:
Because tinyglobby gets passed a pattern and will get you back a file list matching that pattern. It doesn't have a simple match function that matches a path against a glob. But that's what I need here.

Tinyglobby simply doesn't work if you want to have a watch functionality. Chokidar already gives me back the filename. I only need to check if they match. Different use case -> different lib.

Tinyglobby also uses picomatch unter the hood. I wouldn't even save something if I would use tinyglobby instead (which I can't because it doesn't expose a matcher)

Reply:
@Fuzzyma Got it

What about? https://github.com/micromatch/picomatch?tab=readme-ov-file#options

> What do you think about passing parameters here? Before we have them in watchOptions for globs

Some developers may well want to watch dot files for an example (ideally, we should initially have the same values ​​that we had as `chokidar`)

Reply:
I have to look into that.
I couldn't completely figure out the possible things that could be passed here - so I forgot about it 😅.
Maybe you can help me out here by giving me the possible options taht can be passed?

Reply:
Before the globe was removed from chokidar, for example `dot` was part of `watchOptions`, now we have two options - `picomatchOptions` and `chokidarOptions`, I would suggest trying to pass` `watchOptions` to `picomatch` (by index if it is an array)

Reply:
OK I finally got what you were referring to!
I will go over the picomatch options and see if I can make it behave like chokidar v3.

Reply:
Ok, I saw that dotfiles were enabled by default in v3. Besides that, I passed the cwd option to picocolors as well. I am not 100% sure if that is correct behavior but it should. If all paths are relative to the cwd, it should work out. Maybe we need to add a test for that as well.

I also added glob support back for ignored files which I forgot about before